feat: add vue3-vite as supported builder

floroz opened the pull request:
Expand types to support modern `vue3-vite`

tobiasdiez commented:
It's only a type import, so the package strictly speaking doesn't depend on either builder. The point is more that only vite is supported as of now but not webpack. I've added the old vite builder. Do you think my changes are okay?

floroz commented:
> It's only a type import, so the package strictly speaking doesn't depend on either builder. The point is more that only vite is supported as of now but not webpack. I've added the old vite builder. Do you think my changes are okay?

Hey @tobiasdiez, thanks for the explanation.  Yes, the changes look good.

So, based on what you are saying, it won't be a breaking change since we're not replacing one builder (builder-vite) for another, but updating supporting for types of the most recent builder (vue3-vite).

I've updated the commits to reflect that (and remove breaking change annotation).

tobiasdiez commented:
Okay perfect, then let's get this in immediately.
